### PR TITLE
Add info about ranged penalty elimination to archery skill description

### LINKED ISSUE #8398: Archery skill, missing description for shooting without obstacle penalty
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

The Archery skill allows to shoot past castle walls. But the description does not describe this feature.
image

### Describe the possible solution

I suggest adding a description about shooting without obstacle penalty. Players should know about this useful feature.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by zenseii**
Add a description about ranged penalty being when having the archery skill.
Add ranged penalty Get function to `GameStatic` and use in relevant places in the code.

Fix #8398 

image

### CODE REVIEW

File: src/fheroes2/game/game_static.h [hunk 1]
```diff
@@ -63,6 +63,7 @@ namespace GameStatic
     int32_t ObjectVisitedModifiers( const MP2::MapObjectType objectType );
 
     int GetBattleMoatReduceDefense();
+    uint32_t getCastleWallRangedPenalty();
```
Comment (oleg-derevenetz) [inline src/fheroes2/game/game_static.h:66]: Hi @zenseii I usually add a comment to new functions or a method with a brief (if possible) description of what this function does. For example:

```cpp
// Returns the approximate distance between two tiles with given indexes. This distance is calculated as the number of
// tiles (truncated to the nearest smaller integer value) that would need to be traversed in a straight direction to
// overcome the distance that corresponds to the distance that the hero would have to cover when moving between tiles
// with given indexes using the usual rules and penalties.
//
// For example, in order to navigate to a tile B, which is located seven tiles below and five to the right of tile A,
// the hero should move two tiles down and five tiles right down diagonally (diagonal movement costs 50% more), which
// corresponds to 2 + 5 * 1.5 = 9.5 (9 after truncation) tiles traversed in the straight direction.
//
// This function should be avoided unless high precision is not important.
uint32_t GetApproximateDistance( const int32_t pos1, const int32_t pos2 );
```

The advantage is that many IDEs understand such comments and show tooltips when hovering over this function. For example, this is how it looks in MS Visual Studio:

tooltip

The exception is if it is absolutely obvious from the name of the function what it does.

File: src/fheroes2/ai/normal/ai_normal_battle.cpp [hunk 2]
```diff
@@ -951,15 +965,15 @@ namespace AI
                 _myShootersStrength += towerStr;
 
                 if ( !attackerIgnoresCover ) {
-                    _enemyShootersStrength /= 1.5;
+                    _enemyShootersStrength /= 1 + ( GameStatic::getCastleWallRangedPenalty() / 100.0 );
```
Comment (oleg-derevenetz) [inline src/fheroes2/ai/normal/ai_normal_battle.cpp:968]: These `/ 1.5` here and below were introduced in #6743 by replacing `/ 2`. Perhaps the value 1.5 is not directly related to this penalty, but, on the other hand, if the penalty is reduced this coefficient should also be reduced (and if the penalty is increased, it should also be increased). OK, let's do it this way.

**Review by oleg-derevenetz — CHANGES_REQUESTED** [review]
Hi @zenseii I just left one question and one reasoning. Could you please take a look when you have time?

**Review by Branikolog — APPROVED** [review]

**Review by oleg-derevenetz — APPROVED** [review]

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@zenseii , huge thanks for this change!

**Pull request merged**